src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
@hanslovsky ok, doing it with `Function` is fine then. If needed we can add Exceptions later.

Personally I tend to prefer adding a new `@FunctionalInterface` if I for example don't like the method or interface name (For example, `Consumer.accept` is less specific than `FileWriter.store`). You call these things with lambdas most of the time anyways, so mostly the client code looks exactly the same (and if you happen to have a `Consumer` instance `c`, then `c::accept` will give you a `FileWriter` with basically no boilerplate).

In your case `Function.apply` is exactly as unspecific as `NonThrowingCacheLoader.get` would be, though... :-)

(As I said, this is my current personal preference. Maybe I just haven't done this enough yet to run into the downsides...)